Source/WebCore/page/ViewportConfiguration.cpp

```diff
@@ -57,6 +57,16 @@ static inline void adjustViewportArgumentsToAvoidExcessiveZooming(ViewportArgume
     arguments.width = zoomedWidthFromArguments / arguments.zoom;
 }
 
+static inline void ignoreViewportArgumentsToAvoidEnlargedView(ViewportArguments& arguments)
+{
+    if (!viewportArgumentValueIsValid(arguments.width))
+        return;
+
+    if (arguments.width == 1024)
+        arguments.width = 0;
```

Comment:
I wonder if we should set minimumEffectiveDeviceWidth to 1024 here so that devices that are SMALLER than 1024 still use 1024 as the minimum? Should see what @whsieh thinks.

Like, our intent is really to blow out the explicit width on screens that are much bigger, but we don't mean to ever let them shrink *below* 1024.

Reply:
Yeah, that makes sense to me. Or alternately, we could adjust the quirk so it only applies if the real device width (`m_viewLayoutSize`) exceeds 1024.